engine/src/main/java/org/hibernate/search/engine/search/common/spi/SearchIndexValueFieldTypeContext.java

```diff
@@ -49,6 +59,11 @@ default DslConverter<?, F> dslConverter(ValueConvert convert) {
 		switch ( convert ) {
 			case NO:
 				return rawProjectionConverter();
+			case PARSE:
+				throw log.parseConverterNotAllowed( ValueConvert.class.getSimpleName(),
+						ValueConvert.PARSE,
+						List.of( ValueConvert.YES, ValueConvert.NO )
+				);
```

Comment:
There are a few places like this where we need a `FromDocumentValueConverter` instead of a `ToDocumentValueConverter`, which means that the parser doesn't make much sense... 
It means that we cannot pass `ValueConvert.PARSE` to projections, id-match-query, terms-aggregations.

and it works for terms/range/match queries, and sorts

Reply:
Ok but in this specific case it does make sense, no? It would mean parsing the string into an identifier value, then shoving that value through the DSL converter.

Reply:
Hmm sorry I looked at the wrong place.

You're right about this one, but I was talking about that one: https://github.com/hibernate/hibernate-search/pull/4035/files#diff-83eac4231be79581524a307f41fc0d0ac7ff0316ec7662c5d26c20e0222db7c9R37-R41

Reply:
right, with the ID, I wasn't sure, meaning this parser takes a string and transforms it to whatever the thing should be, which in case of an ID is a string? no? so it basically would be string->string, and if we are already passing a string, then why does Search need to "convert" it?

Reply:
Same type doesn't mean same data. The backend-side ID "String" might have a different format than whatever we expect users to provide.

E.g. for an `Instant` identifier one would want to use standard ISO-8601 for parsing, but on the backend side may chose some custom format where special characters are replaced with `-`.

That being said, while in theory this suggestion makes sense, I'm not sure it's easy to implement in practice, as I'm not sure the output of parsing has the same type as the input of DSL converters. Might be worth checking though :shrug:

Reply:
> E.g. for an `Instant` identifier one would want to use standard ISO-8601 for parsing, but on the backend side may chose some custom format where special characters are replaced with `-`.

Another example: parsing may perform some normalization, such as trimming the string before parsing it. In which case `ValueConvert.NO` is not the same as `ValueConvert.PARSE`.

Reply:
ok 😃 thanks! if it's a possible valid use case, then I'll try to make it work 😃👍🏻